interval.hh

```diff
@@ -251,38 +386,78 @@ public:
         return _singular;
     }
     bool is_full() const {
-        return !_start && !_end;
+        return !_start_exists && !_end_exists;
     }
     void reverse() {
         if (!_singular) {
-            std::swap(_start, _end);
+            std::swap(_start_exists, _end_exists);
+            std::swap(_start_inclusive, _end_inclusive);
+            std::swap(_start_value, _end_value);
```

Comment:
It's not safe if !_end_exists since _end_value is not initialized. _start_value may also not exist.